flink-runtime/src/main/java/org/apache/flink/runtime/util/SingleElementIterator.java

```diff
@@ -64,6 +64,7 @@ public void remove() {
 
 		@Override
 		public Iterator<E> iterator() {
+			available = true;
```

Comment:
I think it's better to set `available` to true only if `current` element is not null.

```java
if (current != null) {
  this.available = true;
}
```

Reply:
@wuchong In my opinion, we even need to return a new instance of the iterator and use `set()` method with `current` as the argument, to avoid modifications of the current instance. How do you look at that?